cmake/cmaize/toolchain/toolchain.cmake

```
@@ -60,38 +56,80 @@ cpp_class(Toolchain)
             cpp_raise(FileNotFound "Provided toolchain file does not exist. Toolchain provided: ${toolchain_path}")
         endif()
 
-        Toolchain(SET "${self}" file_path "${toolchain_path}")
-
         # Initialize the toolchain object
         Toolchain(__initialize "${self}")
 
-        # Parse the user toolchain into a list of lines
+        # Read the toolchain file
         file(READ "${toolchain_path}" file_contents)
-        # string(REPLACE "\"" "\\\"" file_contents "${file_contents}")
-        
-        # Parse user options
-        Toolchain(_parse_toolchain "${self}" "${file_contents}")
+
+        # Store toolchain contents
+        cpp_encode_special_chars("${file_contents}" encoded_file_contents)
+        Toolchain(SET "${self}" encoded_toolchain_contents "${encoded_file_contents}")
 
     endfunction()
 
     #[[[
     # Constructor for Toolchain object with a string assumed to be the
     # contents of an already opened toolchain file.
     #
-    # :param toolchain_contents: Contents of the toolchain file.
-    # :type toolchain_contents: str
+    # :param file_contents: Contents of the toolchain file.
+    # :type file_contents: str
     #]]
     cpp_constructor(CTOR Toolchain str)
-    function("${CTOR}" self toolchain_contents)
-
-        # No file provided
-        Toolchain(SET "${self}" file_path "")
+    function("${CTOR}" self file_contents)
 
         # Initialize the toolchain object
         Toolchain(__initialize "${self}")
 
-        # Parse user options
-        Toolchain(_parse_toolchain "${self}" "${toolchain_contents}")
+        # Store toolchain contents
+        cpp_encode_special_chars("${file_contents}" encoded_file_contents)
+        Toolchain(SET "${self}" encoded_toolchain_contents "${encoded_file_contents}")
+
+    endfunction()
+
+    cpp_member(generate_file_contents Toolchain str)
+    function("${generate_file_contents}" self return_value)
+
+        # Initialize the file contents with a warning
+        set(file_string
+            "# This file was generated by CMaize. Any changes made to this
+# file will be overwritten the next time it is generated.\n\n"
+        )
```

FWIW you can do:
```suggestion
        string(CONCAT file_string 
               "# This file was generated by CMaize. Any changes made to this"
               "# file will be overwritten the next time it is generated.\n\n"
        )
```
To break a long string up across multiple lines and preserve indentation.

---

Oh, excellent! I did not know you could do that, but it will clean things up significantly. I'll be sure to do that. Thank you.